File: Backend/AstroGame.Shared/Models/Stellar/BaseTypes/ColonizableStellarObject.cs
using AstroGame.Shared.Models.Stellar.Interfaces;
using System;
using AstroGame.Shared.Models.Players;

namespace AstroGame.Shared.Models.Stellar.BaseTypes
{
    public abstract class ColonizableStellarObject : StellarObject, IColonizable
    {
        protected ColonizableStellarObject()
        {
        }

        protected ColonizableStellarObject(StellarSystem parent) : base(parent)
        {
        }

        public Guid? ColonizedStellarObjectId { get; set; }
        public ColonizedStellarObject ColonizedStellarObject { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/stellar/base_types/colonizable_stellar_object.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/players/colonized_stellar_object.dart';

import 'package:json_annotation/json_annotation.dart';
import 'stellar_object.dart';

@GuidConverter()
abstract class ColonizableStellarObject extends StellarObject {
  // @JsonKey()
  //SingleObjectSystem parentSystem;

  @JsonKey()
  late Guid colonizedStellarObjectId;

  @JsonKey()
  late ColonizedStellarObject? colonizedStellarObject;

  ColonizableStellarObject();
}
